p2p/src/node.rs

```diff
@@ -129,15 +145,20 @@ impl<S: Storage, R: Recon> Drop for Node<S, R> {
 // Allow IntoConnectionHandler deprecated associated type.
 // We are not using IntoConnectionHandler directly only referencing the type as part of this event signature.
 #[allow(deprecated)]
-type NodeSwarmEvent<R> = SwarmEvent<
-            <NodeBehaviour<R> as NetworkBehaviour>::OutEvent,
-            <<<NodeBehaviour<R> as NetworkBehaviour>::ConnectionHandler as IntoConnectionHandler>::Handler as ConnectionHandler>::Error>;
-impl<S: Storage, R: Recon> Node<S, R> {
+type NodeSwarmEvent<IR,MR> = SwarmEvent<
+            <NodeBehaviour<IR,MR> as NetworkBehaviour>::OutEvent,
+            <<<NodeBehaviour<IR,MR> as NetworkBehaviour>::ConnectionHandler as IntoConnectionHandler>::Handler as ConnectionHandler>::Error>;
+impl<S, IR, MR> Node<S, IR, MR>
+where
+    S: Storage,
+    IR: Recon<Key = Interest, Hash = Sha256a>,
+    MR: Recon<Key = EventId, Hash = Sha256a>,
+{
     pub async fn new(
         config: Config,
         rpc_addr: P2pAddr,
         mut keychain: Keychain<S>,
-        recon: Option<R>,
+        recons: Option<(IR, MR)>,
```

Comment:
This seems like it should at least be a type alias. I'm not sure if we'll need to add other recons, but at least that way it's a little easier to adjust.